### PR TITLE
Refactor test orthogonal

### PR DESCRIPTION
**Pull request by Ali-Tehrani**
This pull request refactors the test in orthogonal.py to make the matrix sizes, orthogonal matrices, translation and padding length to be randomized . It generates a random orthogonal matrix via scipy.stats.ortho_group class.

Some tests were removed because they weren't that much different to the previous tests (a7a1b27,  a7e4e72, f9673c4).
Also added "assert(res.s, None)" and removed the dictionary call of results (5361ceb,  67b15b4).
I removed the term unitary from the function document because all unitary matrices here are orthogonal because singular value decomposition of real-valued matrices contain orthogonal matrices and likewise with eigenvalue decomposition of real, symmetric matrices (7c1dc37).
Added a test for incorrect inputs increasing the code coverage (6c3066a).

I will rebase when the previous pull requests goes through, so that it will pass continuous integration.

### CODE REVIEW

**Comment by fwmeng88** [comment]
The failure due to `generic.py` will be fixed after merging PR #70 by replacing `numpy.linalg,pinv` with `scipy.linalg.pinv`. Let's wait for PR #70 to be merged. Thanks for the work!
@Ali-Tehrani

**Review by FarnazH — APPROVED** [review]

**Pull request merged**
